### PR TITLE
Ring constant

### LINKED ISSUE #1766: Follow-up of #1761: use constants from `ring` crate when possible
### Context

As topic, one of the follow-up TODOs of https://github.com/litentry/litentry-parachain/pull/1761

```
// TODO: use constants from `ring` crate, e.g. ring::aead::NONCE_LEN
pub const USER_SHIELDING_KEY_LEN: usize = 32;
pub const USER_SHIELDING_KEY_NONCE_LEN: usize = 12;
pub const USER_SHIELDING_KEY_TAG_LEN: usize = 16;
```

Please notice there might be multiple places where we used these magic numbers

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by songzhouran**

### CODE REVIEW

**Comment by songzhouran** [comment]
close #1766

**Comment by songzhouran** [comment]
No alternative for USER_SHIELDING_KEY_LEN = 32;
There are constants about key_len:
ring::aead::chacha20_poly1305_openssh::KEY_LEN = 64 and ring::aead::chacha::KEY_LEN = 32 is private.

File: primitives/core/Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ version = '0.9.12'
 [dependencies]
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive", "max-encoded-len"] }
 frame-support = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.39", default-features = false }
+ring = { git = "https://github.com/Niederb/ring-xous.git", branch = "0.16.20-cleanup", default-features = false, features = ["alloc"] }
```
Comment (Kailai-Wang) [inline primitives/core/Cargo.toml:10]: any reason not to use the standard `ring`?

File: tee-worker/litentry/primitives/src/aes.rs [hunk 2]
```diff
@@ -20,7 +20,8 @@ extern crate sgx_tstd as std;
 #[cfg(all(not(feature = "std"), feature = "sgx"))]
 extern crate rand_sgx as rand;
 
-use crate::{AesOutput, UserShieldingKeyType, USER_SHIELDING_KEY_NONCE_LEN};
+use crate::{AesOutput, UserShieldingKeyType};
+use ring::aead::NONCE_LEN;
```
Comment (Kailai-Wang) [inline tee-worker/litentry/primitives/src/aes.rs]: You probably want to expose it in `primitives/core/src/key.rs` and import it in this crate and `use` it from the crate, similar to e.g. `AesOutput`

**Review by Kailai-Wang — CHANGES_REQUESTED** [review]

**Comment by Kailai-Wang** [comment]
Sorry, I believe for primitives/core we'd still need a no-std version of `ring` (e.g. `ring-xous`) as it's used in wasm too, see the old problem that we met: https://github.com/litentry/litentry-parachain/issues/963#issuecomment-1325701638

File: tee-worker/litentry/primitives/src/aes.rs [hunk 3]
```diff
@@ -20,9 +20,9 @@ extern crate sgx_tstd as std;
 #[cfg(all(not(feature = "std"), feature = "sgx"))]
 extern crate rand_sgx as rand;
 
-use crate::{
-	AesOutput, UserShieldingKeyNonceType, UserShieldingKeyType, USER_SHIELDING_KEY_NONCE_LEN,
-};
+use crate::NONCE_LEN;
+
+use crate::{AesOutput, UserShieldingKeyNonceType, UserShieldingKeyType};
```
Comment (Kailai-Wang) [inline tee-worker/litentry/primitives/src/aes.rs]: These can be merged

**Review by Kailai-Wang — COMMENTED** [review]
One small thing

**Review by Kailai-Wang — APPROVED** [review]

**Pull request merged**